### PR TITLE
feat: surface secret owner/creator identity from scorer API calls

### PR DESCRIPTION
**Pull request by NextG3nMxJ**
## Summary
- Adds `OwnerInfo` struct to `types.Finding` — captures who owns/created a detected credential
- Populates owner data as a side-effect of existing scorer API calls (no additional network calls)
- AWS: account ID, ARN, IAM username from `sts:GetCallerIdentity`
- GitHub classic PAT: login, email, display name from `GET /user`
- GitLab PAT: username, email, display name from `GET /user`
- Surfaces owner in both JSON output (`"owner": {...}`) and human-readable report (`Owner: alice@corp.com [123456789012]`)
- Engine copies owner from primary match to finding; first-write-wins prevents overwrites

## Test plan
- [x] Unit tests: engine copies owner from match to finding
- [x] Unit tests: does not overwrite existing owner
- [x] Unit tests: nil owner when no condition sets it
- [x] Unit tests: setGitHubOwner/setGitLabOwner helper edge cases
- [x] Full `pkg/...` test suite passes (17 packages)
- [ ] Manual: `titus scan --score-scope` against a repo with known AWS/GitHub/GitLab credentials to verify owner output

Closes LAB-3373

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: cmd/titus/report.go [hunk 1]
```diff
@@ -750,6 +750,19 @@ func outputReportHuman(cmd *cobra.Command, findings []*types.Finding, matches []
 				severityColor.Sprint(f.Score.SuggestedSeverity))
 		}
 
+		if f.Owner != nil {
```
[the remark this answers is not in the document]
Comment (NextG3nMxJ) [reply cmd/titus/report.go:753]: Fixed in b8f297a. Added `owner_json TEXT` column to the findings table (with migration for existing datastores), and wired serialization in `AddFinding`/`GetFindings` so owner survives the store round-trip.

File: pkg/scoring/aws.go [hunk 2]
```diff
@@ -226,11 +226,23 @@ func (c *stsKeyActiveCondition) Evaluate(ctx context.Context, m *types.Match) (b
...
+	m.Owner = &types.OwnerInfo{
+		Service:   "aws",
+		AccountID: awslib.ToString(identity.Account),
+		ARN:       arn,
+	}
+	if username := extractUsernameFromARN(arn); username != "" {
+		m.Owner.User = username
+	} else if roleName := extractRoleNameFromARN(arn); roleName != "" {
+		m.Owner.User = roleName
+	}
```
[the remark this answers is not in the document]
Comment (NextG3nMxJ) [reply pkg/scoring/aws.go]: Fixed in b8f297a. Wrapped the Owner assignment in `if m.Owner == nil`, matching the first-write-wins pattern used in the GitHub and GitLab helpers.

File: pkg/scoring/credit_card.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	return &Scorer{
+		Name:    "ccn-luhn",
+		RuleIDs: []string{"np.ccn.1", "np.ccn.2", "np.ccn.3", "np.ccn.4"},
+		Modifiers: []Modifier{
+			{
+				Name:      "fails-luhn",
+				Priority:  100,
+				Kind:      ModifierKindSetScore,
+				Value:     0,
+				Condition: &ccFailsLuhnCondition{},
```
[the remark this answers is not in the document]
Comment (NextG3nMxJ) [reply pkg/scoring/credit_card.go:70]: This CCN scoring code is from PR #352 (LAB-4104) — it's in this diff because the branches share a base. Score-0 is by design: the Luhn scorer sets score to 0 for failed checksums, and `--min-score` (default 1) filters those from output. This avoids suppressing the finding entirely in case the user wants to inspect all candidates with `--min-score 0`. Deferring to PR #352 for any CCN-specific changes.

File: pkg/rule/rules/ccn.yml [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+  categories: [pii]
+  min_entropy: 2.5
+
+  pattern_requirements:
+    min_digits: 13
+    ignore_if_contains:
+    - "4111111111111111"
+    - "4111-1111-1111-1111"
+    - "4242424242424242"
+    - "4242-4242-4242-4242"
```
[the remark this answers is not in the document]
Comment (NextG3nMxJ) [reply pkg/rule/rules/ccn.yml:31]: CCN rules are from PR #352 (LAB-4104), not this PR. The `stripCardSeparators` function in the Luhn scorer already strips hyphens and spaces before checksum validation. Deferring any CCN rule pattern changes to PR #352.

File: pkg/scoring/credit_card.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	}
+	return sum%10 == 0
+}
+
+// stripCardSeparators removes hyphens and spaces from the captured card group.
+func stripCardSeparators(raw []byte) []byte {
+	out := make([]byte, 0, len(raw))
+	for _, b := range raw {
+		if b != '-' && b != ' ' {
+			out = append(out, b)
```
[the remark this answers is not in the document]
Comment (NextG3nMxJ) [reply pkg/scoring/credit_card.go:35]: Same as above — CCN scoring is from PR #352. The `stripCardSeparators` helper currently strips `-` and space. If we want to handle tabs/newlines from `\s` matches, that's a valid enhancement but belongs in PR #352.

File: pkg/rule/rules/ccn.yml [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+- name: Credit Card Number (Visa)
+  id: np.ccn.1
+  base_score: 25
+  noisy: true
+
+  pattern: |
+    (?x)
+    \b
+    (?P<card> 4[0-9]{3} [\s\-]? [0-9]{4} [\s\-]? [0-9]{4} [\s\-]? [0-9]{4} )
```
[the remark this answers is not in the document]
Comment (NextG3nMxJ) [reply pkg/rule/rules/ccn.yml:19]: CCN rules are from PR #352 (LAB-4104). The rules intentionally use `[0-9]` rather than `\d` since Go's regexp `\d` matches Unicode digits beyond ASCII, which would widen the match unnecessarily. Deferring to PR #352 for any CCN rule discussion.

**Review by michaelweber — APPROVED** [review]
